### PR TITLE
Using implicit solver for soil problems

### PR DESCRIPTION
**Pull request by Zhengyu-Huang**
# Description

Set up an example for using implicit solver for the soil model, this example is in test/Land/Model/haverkamp_test_implicit.jl, which needs to use the implicit solver  [https://github.com/CliMA/ClimateMachine.jl/pull/1483](url)

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by simonbyrne** [comment]
Can you hook it up so it runs as part of the tests?

**Comment by simonbyrne** [comment]
@Zhengyu-Huang Can you add it to buildkite?

**Comment by Zhengyu-Huang** [comment]
> @Zhengyu-Huang Can you add it to buildkite?

Sure

File: test/Land/Model/haverkamp_implicit_test.jl
Comment (kmdeck) [inline test/Land/Model/haverkamp_implicit_test.jl:122]: Let's remove these lines

File: test/Land/Model/haverkamp_implicit_test.jl
Comment (kmdeck) [inline test/Land/Model/haverkamp_implicit_test.jl:179]: I think we can delete, as Q defined above?

File: test/Land/Model/haverkamp_implicit_test.jl
Comment (kmdeck) [inline test/Land/Model/haverkamp_implicit_test.jl:170]: I thought vdg was tendency the implicit solver was used with, and dg-vdg would use the explicit solver. Is that true in general, but here, since we only have one column, you can set split_explicit_implicit = false?

**Review by kmdeck — COMMENTED** [review]
looks good to me, left a couple minor comments. before merging, I think it is convention to squash your commits. The beginning of this goes over that.
https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Review by kmdeck — APPROVED** [review]
approving so you can submit when you want! but should squash commits first I think.

**Comment by jkozdon** [comment]
> but should squash commits first I think.

Yes. Please do.

File: test/Land/Model/haverkamp_implicit_test.jl
Comment (Zhengyu-Huang) [reply test/Land/Model/haverkamp_implicit_test.jl:170]: " I thought vdg was tendency the implicit solver was used with, and dg-vdg would use the explicit solver. Is that true in general, but here, since we only have one column ", yes
my understanding of "split_explicit_implicit = false",  we use dg-vdg as explicit part instead of defining it https://clima.github.io/ClimateMachine.jl/latest/APIs/Numerics/ODESolvers/ODESolvers/#ClimateMachine.ODESolvers.AdditiveRungeKutta

**Comment by Zhengyu-Huang** [comment]
> looks good to me, left a couple minor comments. before merging, I think it is convention to squash your commits. The beginning of this goes over that.
> https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

I think I messed this up with squash , should I close this PR and do new PR ?

**Pull request closed without merging**
